src/plugins/data-card-tool/data-card-tool.tsx

```diff
@@ -139,14 +149,35 @@ export const DataCardToolComponent: React.FC<IToolTileProps> = observer((props)
 
   const addCardClasses = classNames("add-card", "teal-bg", { hidden: !shouldShowAddCase });
   const removeCardClasses = classNames("remove-card", { hidden: !shouldShowDeleteCase });
-  const toolbarProps = useToolbarToolApi({ id: model.id, enabled: !readOnly, onRegisterToolApi, onUnregisterToolApi });
 
-  return (
-    <div className="data-card-tool">
-      <DataCardToolbar model={model} documentContent={documentContent} toolTile={toolTile}
-                      currEditAttrId={currEditAttrId}
-                      setImageUrlToAdd={setImageUrlToAdd} {...toolbarProps} />
+  const toolbarProps = useToolbarToolApi(
+    {
+      id: model.id,
+      enabled: !readOnly, // "enabled" is "visible"
+      onRegisterToolApi,
+      onUnregisterToolApi
+    }
+  );
+
+  const handleToolClick = (event: React.MouseEvent<HTMLDivElement | HTMLInputElement>) => {
+    const targetedElement = event.target as HTMLInputElement;
+    const isUploader = targetedElement.classList.contains("input-for-upload");
+    if ( isUploader ) return;
```

Comment:
This is an improvement but still seems like it's a bit ad hoc. I played around with this a bit and will add a commit with an alternative approach.

Reply:
For some reason there is no space to comment on the styling note above.  The reason I used `97%` is that at 100% height the input just _slightly_ covers up the borders, making the borders on the value side thinner than the borders on the label side.